web/pandas/pdeps/0014-string-dtype.md

```diff
@@ -0,0 +1,373 @@
+a "fallback" option in case PyArrow is not installed. The original `StringDtype`
+backed by a numpy object-dtype array of Python strings can be mostly reused for
+this (adding a new variant of the dtype) and a new `StringArray` subclass only
+needs minor changes to follow the above-mentioned missing value semantics
+([GH-58451](https://github.com/pandas-dev/pandas/pull/58451)).
+
+For pandas 3.0, this is the most realistic option given this implementation has
+already been available for a long time. Beyond 3.0, further improvements such as
+using NumPy 2.0 ([GH-58503](https://github.com/pandas-dev/pandas/issues/58503))
+or nanoarrow ([GH-58552](https://github.com/pandas-dev/pandas/issues/58552)) can
+still be explored, but at that point that is an implementation detail that
+should not have a direct impact on users (except for performance).
+
+For the original variant of `StringDtype` using `pd.NA`, currently the default
+storage is `"python"` (the object-dtype based implementation). Also for this
+variant, it is proposed follow the same logic for determining the default
+storage, i.e. the default to `"pyarrow"` if available, and otherwise
+fall back to `"python"`.
+
+### Naming
+
+Given the long history of this topic, the naming of the dtypes is a difficult
+topic.
+
+In the first place, it should be acknowledged that most users should not need to
+use storage-specific options. Users are expected to specify a generic name (such
+as `"str"` or `"string"`), and that will give them their default string dtype
+(which depends on whether PyArrow is installed or not).
+
+For the generic string alias to specify the dtype, `"string"` is already used
+for the `StringDtype` using `pd.NA`. This PDEP proposes to use `"str"` for the
+new default `StringDtype` using `NaN`. This ensures backwards compatibility for
+code using `dtype="string"`, and was also chosen because `dtype="str"` or
+`dtype=str` currently already works to ensure your data is converted to
+strings (only using object dtype for the result).
+
+But for testing purposes and advanced use cases that want control over the exact
+variant of the `StringDtype`, we need some way to specify this and distinguish
+them from the other string dtypes.
+
+Currently (pandas 2.2), `StringDtype(storage="pyarrow_numpy")` is used for the new variant using `NaN`,
+where the `"pyarrow_numpy"` storage was used to disambiguate from the existing
+`"pyarrow"` option using `pd.NA`. However, `"pyarrow_numpy"` is a rather confusing
+option and doesn't generalize well. Therefore, this PDEP proposes a new naming
+scheme as outlined below, and `"pyarrow_numpy"` will be deprecated and removed
+before pandas 3.0.
+
+The `storage` keyword of `StringDtype` is kept to disambiguate the underlying
+storage of the string data (using pyarrow or python objects), but an additional
+`na_value` is introduced to disambiguate the the variants using NA semantics
```

Comment:
The only thing with `na_value` I wonder about is if its worth specifying `na_value=np.nan` or instead using something more generic like `na_handling="legacy"`

The advantage to the latter is it would work with any type, especially as we think about moving towards the logical type system. `pd.DatetimeType(na_value=np.nan)` probably should not exist, so a user would either have to opt in to `pd.DatetimeType(na_value=pd.NaT)` explicitly or gloss over that and just have them use `pd.XXXType(na_handling="legacy")` when they want historical pandas behavior

Reply:
While it might be a bit more future looking, I personally wouldn't want to call the default behaviour "legacy". That seems like a strange message, because it will potentially only be legacy in 4.0 (or in later 3.x releases if we provide an option to globally opt in to using `pd.NA`). But for most users for 3.0 it is simply the default missing value handling.

Reply:
Fair point. Not tied to the name. Maybe "hybrid" or "mixed"?

Reply:
I would personally prefer to keep `na_value` for this proposal. On the short term, that's the most minimal change (the attribute already exists, so we are not adding a new attribute, only exposing it as keyword as well).

We can certainly think of a better or more generic way to handle this in the context of PDEP-13. And if we do so, I think it will be very easy to keep compatibility with the `na_value` keyword just in `StringDtype` (it will be easy to still accept that on the short term and translate that to the proper new keyword), i.e. not much maintenance overhead to keep that working for StringDtype to avoid code churn for users.

(any keyword that we will be adding related to the NA handling will (I hope) be temporary anyway, so also a `na_handling` keyword might at some point get deprecated again)

Reply:
Fair enough